scripts/globus/create_guest_collection.py

```diff
@@ -151,23 +151,123 @@
 
 # Get the mapped collection id
 scope.add_dependency(scopes.GCSCollectionScopeBuilder(mapped_collection_id).data_access)
-authorizer = globus_sdk.ClientCredentialsAuthorizer(confidential_client, scopes=scope)
-client = globus_sdk.GCSClient(DATAFED_GCS_URL, authorizer=authorizer)
+final_authorizer = globus_sdk.ClientCredentialsAuthorizer(confidential_client, scopes=scope)
+final_client = globus_sdk.GCSClient(DATAFED_GCS_URL, authorizer=final_authorizer)
 
 guest_collection_found = False
+guest_collection_id = None
+existing_base_path = None
 
 for item in collection_list["data"]:
     print(item["display_name"])
     if item["display_name"] == guest_collection_name:
         guest_collection_found = True
         guest_collection_id = item["id"]
+        # The collection list doesn't contain base path information
+        # We'll need to get it from collection details
+        existing_base_path = None
+        print(f"Found existing guest collection {guest_collection_id}")
+        print(f"Collection data keys: {list(item.keys())}")
         break
 
-if guest_collection_found:
-    print(f"Removing current guest collection {guest_collection_id}")
-    response = client.delete_collection(guest_collection_id)
+def should_recreate_guest_collection(collection_id, current_base_path, cache_directory):
+    """
+    Determine if the guest collection should be recreated based on base path changes.
+    
+    Args:
+        collection_id: ID of the existing guest collection
+        current_base_path: Current base path from environment
+        cache_directory: Directory for cache files
+    
+    Returns:
+        tuple: (should_recreate: bool, reason: str)
+    """
+    # Normalize base paths to handle edge cases
+    current_base_path = current_base_path.rstrip('/') if current_base_path != '/' else '/'
+    
+    # Get cached base path
+    cached_base_path = utils.get_cached_base_path(collection_id, cache_directory)
+    
+    if cached_base_path:
+        cached_base_path = cached_base_path.rstrip('/') if cached_base_path != '/' else '/'
+        print(f"Found cached base path: '{cached_base_path}'")
+    else:
+        print("No cached base path found")
+    
+    print(f"Current BASE_PATH environment variable: '{current_base_path}'")
+    
+    # Check for manual override first
+    force_recreate = os.getenv("DATAFED_FORCE_RECREATE_GUEST_COLLECTION", "false").lower() == "true"
+    if force_recreate:
+        return True, "DATAFED_FORCE_RECREATE_GUEST_COLLECTION is set, forcing recreation"
+    
+    # Compare base paths
+    if cached_base_path is None:
+        return False, "No cached base path available, keeping existing collection to avoid unnecessary recreation"
```

Comment:
Wouldn't you want to recreate by default here? If the base path file is not available you don't know if the base path is the same, the only way to guarantee that is to recreate it.